src/docfx/build/markdown/BuildMarkdown.cs

```diff
@@ -1,17 +1,32 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Threading.Tasks;
 
 namespace Microsoft.Docs.Build
 {
     internal static class BuildMarkdown
     {
-        public static Task Build(Context context, Document file, TableOfContentsMap tocMap)
+        public static Task Build(Context context, Document file, TableOfContentsMap tocMap, Action<Document> buildChild)
         {
             var markdown = file.ReadText();
 
-            var (html, markup) = MarkdownUtility.Markup(markdown, file, context);
+            var (html, markup) = MarkdownUtility.Markup(
+                markdown,
+                file,
+                context,
+                (a, b, c) =>
+                {
+                    // resolve href link
+                    var (link, buildItem) = Resolve.TryResolveHref(a, b, c);
+                    if (buildItem != null)
+                    {
+                        buildChild(buildItem);
+                    }
+
```

Comment:
Is this for single file build? Seems like a separate story we will support later, not needed for normal build.

Reply:
it's for single file build including it's dependency

---
In reply to: [191305652](https://github.com/dotnet/docfx/pull/2818#discussion_r191305652) [](ancestors = 191305652)